### PR TITLE
Add synonyms

### PR DESCRIPTION
**Pull request by maykcaldas**

### CODE REVIEW

**Comment by maykcaldas** [comment]
Implemented the method. Still need to remove the hard-coded file address, write tests and include this in the usage notebook.

**Comment by GiaJordan** [comment]
> Implemented the method. Still need to remove the hard-coded file address, write tests and include this in the usage notebook.

@maykcaldas got it, let me know when you're ready for me to take a look

**Comment by maykcaldas** [comment]
@GiaJordan 
Can you have a quick look to suggest how I can test it? Because we're not uploading the synonyms list, so we can't run the new method in the workflow

**Comment by GiaJordan** [comment]
@maykcaldas in its current form it wouldn't be the easiest to test. We could create a dummy test `.csv` file with synonyms and pass that path in if we wanted to, but I don't think that's what we've done elsewhere in this codebase. 

What we have done is write tests that use the `.gz` file and only run them locally, and then skip them on gh actions. This isn't the best but it'll work for now, and we could go back later and refactor so it's easier to test

**Comment by maykcaldas** [comment]
Great, @GiaJordan
I implemented a test to run locally. Please, feel free to review it when you have time.

File: tests/test_dataset_builder.py [hunk 1]
```diff
@@ -125,4 +131,17 @@ def test_clean_dataset(self, dataset_df, pug_view_page_one, use_text):
             assert (DB.dataset['AllText'].notna()).all()
         assert (DB.dataset['NumAtoms'].notna()).all()
         assert len(DB.dataset) < orginal_length
-        
\ No newline at end of file
+        
+
+    @pytest.mark.skipif(IN_GITHUB_ACTIONS, reason="Synonyms file is not available in GitHub")
```
Comment (GiaJordan) [inline tests/test_dataset_builder.py:136]: Test looks good, could we just update the mark on `test_instantiate_DB_from_file` to match this one as well?

**Review by GiaJordan — CHANGES_REQUESTED** [review]

**Review by GiaJordan — APPROVED** [review]

**Pull request merged**